src/main/java/com/example/bookstoreapp/controller/CategoryController.java

```diff
@@ -0,0 +1,69 @@
+package com.example.bookstoreapp.controller;
+
+import com.example.bookstoreapp.dto.BookDto;
+import com.example.bookstoreapp.dto.CategoryDto;
+import com.example.bookstoreapp.dto.CreateCategoryDto;
+import com.example.bookstoreapp.servive.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Categories management", description = "Endpoints for management categories")
+@RestController
+@RequiredArgsConstructor
+@PreAuthorize("hasRole('ADMIN')")
```

Comment:
Why this parameter set on class level? Does it mean that only admin can access categories related endpoint?

Reply:
My bad.